java/vector/src/main/java/org/apache/arrow/vector/BufferBacked.java

```diff
@@ -28,6 +28,8 @@
 
   void load(ArrowFieldNode fieldNode, ArrowBuf data);
 
+  void load(ArrowBuf data);
```

Comment:
could you add javadoc to clarify the difference between the two load methods.
In particular if we can load without the FieldNode, what do we do differently when we have the FieldNode?